### ISSUE TITLE
Will it run on PHP 8.0?

### ISSUE THREAD

**Issue by simonwes** [mikespub-org/rvolz-BicBucStriim#5]
The SynoCommunity package requests a PHP 8.2 package which is not available on Synology DSM 7.1. 
PHP 8.0 is the highest PHP version there. Older Diskstations don't get newer DSM releases. anymore and by that no PHP > 8.0.

When using the download package and trying to run it under PHP 8.0 I see the message
Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 8.2.0".

Is there a chance to run this under PHP 8.0?

**Comment by mikespub** [comment]
I'm not quite sure how the dependencies work out between DSM, Apache and package versions created by SynoCommunity, but in general BicBucStriim 3.x versions rely on PHP >= 8.1 for early versions and PHP >= 8.2 in more recent ones.

The older BicBucStriim releases like 1.7.2 should work with PHP 8.0, but you may need to download the [source code](https://github.com/mikespub-org/rvolz-BicBucStriim/releases/tag/v1.7.2) and run `composer update -o` yourself to get compatible vendor packages.

Does that help?

**Comment by mikespub** [comment]
Or if you have Web Station up and running, since BBS is simply PHP code that could run anywhere after installation, nothing stops you from downloading the .zip file here yourself and extract it under /volume1/web/bbs for example, run composer to install dependencies and then adapt the permissions for the Calibre directory yourself.

That way you don't need the right version of SynoCommunity package, but it requires shell access to your NAS...

**Comment by mikespub** [comment]
I updated [README](https://github.com/mikespub-org/rvolz-BicBucStriim/blob/main/README.md) to clarify the use of release packages, and renamed all release packages with the (minimum) -phpxx version they were prepared with.

The packages jumped from PHP 7.4 for v1.7.2 to PHP 8.2 for v3.3.0, but that doesn't mean v3.x.x can't run with PHP 8.0 at all - it simply isn't packaged for it...

**Comment by simonwes** [comment]
Thanks for quick reply. I'm not a PHP developer. But I managed to run the master branch on a fresh Debian 12 VM with PHP 8.0. I had to tweak some versions in composer.json as you mentioned:

```
-        "php": ">=8.2",
+        "php": ">=8.0",
-        "psr/cache": "^3.0",
+        "psr/cache": "^1.0",
-        "8ctopus/apix-log": "^3.0",
+        "8ctopus/apix-log": "^2.1.0",
-        "middlewares/cache": "^3.0",
-        "symfony/cache": "^7.1"
+        "middlewares/cache": "^2.1.0",
+        "symfony/cache": "^5.4"
-        "phpunit/phpunit": "^11.4"
+        "phpunit/phpunit": "^9" 
``` 

From a first look it seems to work well with a copy of my Calibre lib.

**Comment by mikespub** [comment]
Yep, that looks about the same as the issues I found ;-)

Besides potential breaking changes between major releases, there were some weird inter-dependencies with caching that made me simply say: let's forget about the whole thing, and stick to PHP 8.2 for future maintenance & development...

Did you try this with the v3.5.2 version or some earlier one?

**Comment by simonwes** [comment]
I just cloned the repo yesterday and used your main branch. I think that is 3.5.2?

**Comment by mikespub** [comment]
Yep, that's the one - happy to hear that still works for you then :-)

So I'm not going to maintain 2 versions of composer.json or support PHP 8.0, which is [EOL with no security fixes since December 2023](https://www.php.net/supported-versions.php), but I'll try to avoid introducing any incompatible code changes in BicBucStriim itself if I can.

But if/when the [Slim PHP framework](https://www.slimframework.com/) on which it runs comes out with version 5.0, all bets are off :-)
